lib/common/hostinfo.c

```diff
@@ -50,6 +51,31 @@ static struct {
 
 size_t h2o_hostinfo_max_threads = 1;
 
+static const char *dns_rcode_str_from_gai_error(int ret)
+{
+    switch (ret) {
+    case EAI_NONAME: /* The node or service is not known. */
+        return "NXDOMAIN";
+
+    case EAI_ADDRFAMILY: /* The specified network host does not have any network addresses in the requested address family. */
+    case EAI_NODATA: /* The specified network host exists, but does not have any network addresses defined. */
+        return "NODATA";
+
+    case EAI_FAIL: /* The name server returned a permanent failure indication. */
+        return "REFUSED";
+
+    case EAI_AGAIN: /* The name server returned a temporary failure indication.  Try again later. */
+    case EAI_BADFLAGS: /* hints.ai_flags contains invalid flags; or, hints.ai_flags included AI_CANONNAME and name was NULL. */
+    case EAI_FAMILY: /* The requested address family is not supported. */
+    case EAI_MEMORY: /* Out of memory. */
+    case EAI_SERVICE: /* The requested service is not available for the requested socket type. */
+    case EAI_SOCKTYPE: /* The requested socket type is not supported. */
+    case EAI_SYSTEM: /* Other system error, check errno for details. */
+    default:
+        return "SERVFAIL";
```

Comment:
All proxy-status "parameters" are optional and we can add custom ones, which means that in dns error cases that do not clearly translate to and "rcode", we have the option to omit it.  We are already populating the "details" parameter with the text, so we can continue to do that.  Rather than mapping a bunch of miscellaneous cases to "rcode=SERVFAIL", I can make those cases omit the "rcode" parameter, if you agree with that plan.  What do you think?

Reply:
Works for me. My concern would be resolved if we omit rcode when uncertain.

Reply:
Done in 4675191dd.